enrichments/trace/internal/elastic/span_test.go

```diff
@@ -0,0 +1,447 @@
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package elastic
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	semconv "go.opentelemetry.io/collector/semconv/v1.25.0"
+	"google.golang.org/grpc/codes"
+)
+
+// Tests the enrichment logic for elastic's transaction definition.
+func TestElasticTransactionEnrich(t *testing.T) {
+	for _, tc := range []struct {
+		name          string
+		input         ptrace.Span
+		enrichedAttrs map[string]any
+	}{
+		{
+			name:  "empty",
+			input: ptrace.NewSpan(),
+			enrichedAttrs: map[string]any{
+				AttributeEventOutcome:      "success",
+				AttributeTransactionResult: "Success",
+				AttributeTransactionType:   "unknown",
+			},
+		},
+		{
+			name: "http_status_ok",
+			input: func() ptrace.Span {
+				span := ptrace.NewSpan()
+				span.Attributes().PutInt(semconv.AttributeHTTPStatusCode, http.StatusOK)
+				return span
+			}(),
+			enrichedAttrs: map[string]any{
+				AttributeEventOutcome:      "success",
+				AttributeTransactionResult: "HTTP 2xx",
+				AttributeTransactionType:   "request",
+			},
+		},
+		{
+			name: "http_status_1xx",
+			input: func() ptrace.Span {
+				span := ptrace.NewSpan()
+				// attributes should be preferred over span status
+				span.Status().SetCode(ptrace.StatusCodeOk)
+				span.Attributes().PutInt(
+					semconv.AttributeHTTPResponseStatusCode,
+					http.StatusProcessing,
+				)
+				return span
+			}(),
+			enrichedAttrs: map[string]any{
+				AttributeEventOutcome:      "success",
+				AttributeTransactionResult: "HTTP 1xx",
+				AttributeTransactionType:   "request",
+			},
+		},
+		{
+			name: "http_status_5xx",
+			input: func() ptrace.Span {
+				span := ptrace.NewSpan()
+				span.Attributes().PutInt(
+					semconv.AttributeHTTPStatusCode, http.StatusInternalServerError)
+				return span
+			}(),
+			enrichedAttrs: map[string]any{
+				AttributeEventOutcome:      "failure",
+				AttributeTransactionResult: "HTTP 5xx",
+				AttributeTransactionType:   "request",
+			},
+		},
+		{
+			name: "grpc_status_ok",
+			input: func() ptrace.Span {
+				span := ptrace.NewSpan()
+				// attributes should be preferred over span status
+				span.Status().SetCode(ptrace.StatusCodeOk)
+				span.Attributes().PutInt(
+					semconv.AttributeRPCGRPCStatusCode,
+					int64(codes.OK),
+				)
+				return span
+			}(),
+			enrichedAttrs: map[string]any{
+				AttributeEventOutcome:      "success",
+				AttributeTransactionResult: "OK",
+				AttributeTransactionType:   "request",
+			},
+		},
+		{
+			name: "grpc_status_internal_error",
+			input: func() ptrace.Span {
+				span := ptrace.NewSpan()
+				// attributes should be preferred over span status
```

Comment:
This comment is a bit confusing to me: I think what you're saying is that the status code takes precedence over span status for setting the `transaction.result`? But OTOH it's the opposite for `event.outcome`. Maybe clarify that in the comment?

Reply:
Fixed! Thanks for finding that!